sse2neon.h

```diff
@@ -2357,6 +2357,77 @@ FORCE_INLINE __m128 _mm_setzero_ps(void)
     return vreinterpretq_m128_f32(vdupq_n_f32(0));
 }
 
+FORCE_INLINE __m64 _mm_shuffle_pi16(__m64 a, int imm)
+{
+    int16x4_t ret = vdup_n_s16(0);
+    int16_t a0 = vget_lane_s16(vreinterpret_s16_m64(a), 0);
+    int16_t a1 = vget_lane_s16(vreinterpret_s16_m64(a), 1);
+    int16_t a2 = vget_lane_s16(vreinterpret_s16_m64(a), 2);
+    int16_t a3 = vget_lane_s16(vreinterpret_s16_m64(a), 3);
+
+    switch (imm & 0x3) {
```

Comment:
If `__builtin_shufflevector` is available, use `__builtin_shufflevector` to implement this intrinsic.

Reply:
added